contracts/staking/MasterChef.sol

```diff
@@ -236,26 +237,28 @@ contract MasterChef is Ownable {
     function deposit(address _token, uint256 _amount) external {
         PoolInfo storage pool = poolInfo[_token];
         require(pool.lastRewardTime > 0);
-        UserInfo storage user = userInfo[_token][msg.sender];
         _updateEmissions();
         _updatePool(_token, totalAllocPoint);
-        if (user.amount > 0) {
-            uint256 pending =
-                user.amount.mul(pool.accRewardPerShare).div(1e12).sub(
-                    user.rewardDebt
-                );
-            _mint(msg.sender, pending);
+        UserInfo storage user = userInfo[_token][msg.sender];
+        uint256 amount = user.amount;
+        uint256 accRewardPerShare = pool.accRewardPerShare;
+        if (amount > 0) {
+            uint256 pending = amount.mul(accRewardPerShare).div(1e12).sub(user.rewardDebt);
+            if (pending > 0) {
+                userBaseClaimable[msg.sender] = userBaseClaimable[msg.sender].add(pending);
+            }
         }
         IERC20(_token).safeTransferFrom(
             address(msg.sender),
             address(this),
             _amount
         );
-        user.amount = user.amount.add(_amount);
-        user.rewardDebt = user.amount.mul(pool.accRewardPerShare).div(1e12);
+        amount = amount.add(_amount);
```

Comment:
Use of `amount` and `_amount` here is ambiguous. Can we rename `amount` to `userAmount`?